lib/log/logger.ts

```diff
@@ -27,75 +26,96 @@ function mkdirSync(path: string): void {
         }
     }
 }
-
 interface LoggerOptions {
     fluentD?: {
         host?: string
         port?: number
         timeout?: number
         tag?: string
     },
+    format?: {
+        timestamp?: boolean,
+        json?: boolean,
+        colorize?: boolean
+    },
     cwd?: string
     logDirectory?: string
 }
+// default format for Console, error at winston: if not json is specified is not logging in console
+export const LoggerFormat = format.printf(({level, message }) => {
+    return `${level}: ${message}`;
+});
+export const Logger = (options: LoggerOptions) => {
 
-export class Logger extends winston.Logger {
-
-    constructor(options: LoggerOptions) {
-        options = _.defaultsDeep(options || {}, {
-            fluentD: {},
-            logDirectory: null,
-            cwd: process.cwd()
-        });
-
-        let transports: TransportInstance[] = [
-            new winston.transports.Console({
-                colorize: true
-            })
-        ];
+    options = _.defaultsDeep(options || {}, {
+        fluentD: {},
+        logDirectory: null,
+        format: {
+            timestamp: true,
+            json: false,
+            colorize: false
+        },
+        cwd: process.cwd()
+    });
 
-        if (options.logDirectory) {
-            // Create the log directory if it doesn't exist
-            mkdirSync(options.logDirectory);
+    // Defining custom formats 
+    const styleFormats = [];
 
-            transports.push(new winston.transports.File({
-                filename: path.resolve(options.logDirectory, 'app.log'),
-                timestamp: true,
-                json: false,
-                maxfiles: 5,
-                maxsize: 10485760,
-                level: 'info'
-            }));
+    // special case: if json is not defined , use by default the custom format 
+    if (options.format.json === false) {
+        // going to global
+        styleFormats.push( global.LabShare.LoggerFormat);
```

Comment:
Accessing a global here will break the Hedwig Storage api project which doesn’t use LSC as a CLI.